### PR TITLE
feat(analytics): make analytics-metric-provider public [MA-1794]

### PR DESCRIPTION
**Pull request by adorack**
# Summary

## PR Checklist

* [ ] **Naming & Structure:** the files and package structure use the conventions outlined in the [Creating a Package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md).
* [ ] **Tests pass:** check the output of all package unit and/or component tests.
  * [ ] If this PR is the result of a bug, test coverage was added accordingly.
* [ ] **Functional:** all changes do not break existing APIs, but if so, a BREAKING CHANGE commit is in place to bump the major version.
* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/public-ui-components#committing-changes).
* [ ] **Docs:** includes a technically accurate README, and the docs have been updated accordingly based on the changes in this PR.

### CODE REVIEW

File: docs/moving-packages.md
Comment (adamdehaven) [inline docs/moving-packages.md:100]: Thank you! 

Would you mind creating a PR in `shared-ui-components` with this docs addition as well?

File: packages/analytics/analytics-metric-provider/CHANGELOG.md
Comment (adamdehaven) [inline packages/analytics/analytics-metric-provider/CHANGELOG.md:1]: nit: I'd recommend deleting this file completely and it will be auto-generated

File: packages/analytics/analytics-metric-provider/package.json
Comment (adamdehaven) [inline packages/analytics/analytics-metric-provider/package.json:53]: issue: i18n should be moved to `devDependencies` and `peerDependencies`

File: packages/analytics/analytics-metric-provider/vite.config.ts
Comment (adamdehaven) [inline packages/analytics/analytics-metric-provider/vite.config.ts:19]: After moving `i18n` package to peerDependencies, you should externalize the dependency
```suggestion
  build: {
    lib: {
      // The kebab-case name of the exposed global variable. MUST be in the format `kong-ui-public-{package-name}`
      // Example: name: 'kong-ui-public-demo-component'
      name: `kong-ui-public-${sanitizedPackageName}`,
      entry: resolve(__dirname, './src/index.ts'),
      fileName: (format) => `${sanitizedPackageName}.${format}.js`,
    },
    rollupOptions: {
      // Make sure to externalize deps that shouldn't be bundled into your library
      external: ['@kong-ui-public/i18n'],
      output: {
        // Provide global variables to use in the UMD build for externalized deps
        globals: {
          '@kong-ui-public/i18n': 'kong-ui-public-i18n',
        },
      },
    },
  },
```

File: docs/moving-packages.md
Comment (adorack) [reply docs/moving-packages.md:100]: Sure, no problem.

File: packages/analytics/analytics-metric-provider/CHANGELOG.md
Comment (adorack) [reply packages/analytics/analytics-metric-provider/CHANGELOG.md]: Ah, I wasn't sure -- done.

File: packages/analytics/analytics-metric-provider/package.json
Comment (adorack) [reply packages/analytics/analytics-metric-provider/package.json]: Done.

File: packages/analytics/analytics-metric-provider/vite.config.ts
Comment (adorack) [reply packages/analytics/analytics-metric-provider/vite.config.ts]: Done.

**Review by adamdehaven — APPROVED** [review]
LGTM but @adorack please defer to your team for final approval before merging

**Review by hfukada — APPROVED** [review]

**Pull request merged**
